### ISSUE TITLE
Variant with bounding volume hierarchy such as a k-d tree?

### ISSUE THREAD

**Issue by Qqwy** [mattgodbolt/pt-three-ways#5]
I very much liked the talk.
I would be really interested in how these three techniques would compare performance-wise, if a bounding volume hierarchy, like a k-d tree were to be introduced.
My expectation is that all three versions will speed up dramatically; however I wonder which one of the three is able to benefit from it the most.

**Comment by mattgodbolt** [comment]
Absolutely: it was always my intention to add this after to see how easy it was to add to the three approaches. In the end I didn't have time, and I wanted to get to the bottom of the performance issues found in the DoD version.
